app/kuma-dp/pkg/dataplane/metrics/server.go

```diff
@@ -208,25 +261,26 @@ func (s *Hijacker) getStats(ctx context.Context, initReq *http.Request, app Appl
 	}
 	if err != nil {
 		logger.Error(err, "failed call", "name", app.Name, "path", app.Path, "port", app.Port)
-		return nil
+		return nil, ""
 	}
 
+	respContentType := resp.Header.Get("content-type")
```

Comment:
Do we need to parse this to compare the `content-type` or are they guaranteed to be exactly equal to the `expfmt` values?

Reply:
The content-types returned by the applications are expected to be equal to the `expfmt` value, but we can not guarantee though. But either way, the hijacker needs to know the content-type returned by all the apps that it is aggregating the metrics of. Then only it can decide the best possible content-type that needs to be sent back to the prometheus scraper.

Reply:
I was just wondering whether it'd be better to use something like https://pkg.go.dev/mime#ParseMediaType to compare them.

Reply:
We're still just comparing on exact string equality. Are Prometheus metrics endpoints guaranteed to use this exact string as opposed to some set of equivalent content-types?

Reply:
When we instrument our applications to export metrics in Prometheus format, we usually rely on 3rd party libraries and they adhere to Prom standard content-types. So yes, it's expected the metrics endpoints to return the exact same content-types (string), else the prom scraper won't be able to process the response.
I checked `expfmt` package. They have `expfmt.ResponseFormat()` to parse content-type of the application response. This ReseponseFormat() internally uses `mime.ParseMediaType()` that you suggested earlier to parse the content-types. But unfortunately it does not support the latest openmetrics type, So implemented this way.

Reply:
Is the order of parameters guaranteed? I mean what if an application sends `application/openmetrics-text; charset=utf-8; version=0.0.1` instead of `application/openmetrics-text; version=0.0.1; charset=utf-8`?

Reply:
As far as the standard 3rd party libraries are used, I have seen the params in order or at least it's expected to be in order. I am not sure if Prometheus scraper handles this scenario. But it does not matter if Prometheus scraper handles it or not, as the W3 content-type header spec does not enforce any ordering, our Metric Hijacker should handle it.
This is a nice catch. Let me handle this scenario as well.